src/extension.ts

```diff
@@ -26,60 +27,84 @@ export function activate(context: vscode.ExtensionContext) {
         if (node) {
             outputChannel.appendLine(`Stopping App "${node.site.name}"...`);
-            node.stop(azureAccount).then(() => outputChannel.appendLine(`App "${node.site.name}" has been stopped.`), err => outputChannel.appendLine(err));
+            await node.stop(azureAccount).then(() => outputChannel.appendLine(`App "${node.site.name}" has been stopped.`), err => outputChannel.appendLine(err));
         }
-    }));
-    context.subscriptions.push(vscode.commands.registerCommand('appService.Restart', (node: AppServiceNode) => {
+    });
+    initAsyncCommand(context, 'appService.Restart', async (node: AppServiceNode) => {
         if (node) {
             outputChannel.appendLine(`Restarting App "${node.site.name}"...`);
-            node.restart(azureAccount).then(() => outputChannel.appendLine(`App "${node.site.name}" has been restarted.`), err => outputChannel.appendLine(err));
+            await node.restart(azureAccount).then(() => outputChannel.appendLine(`App "${node.site.name}" has been restarted.`), err => outputChannel.appendLine(err));
         }
-    }));
-    context.subscriptions.push(vscode.commands.registerCommand('appService.CreateWebApp', async () => {
+    });
+    initAsyncCommand(context, 'appService.CreateWebApp', async () => {
         const wizard = new WebAppCreator(outputChannel, azureAccount);
         const result = await wizard.run();
         
         if (result.status === 'Completed') {
             vscode.commands.executeCommand('appService.Refresh');
         }
-    }));
-    context.subscriptions.push(vscode.commands.registerCommand('appService.DeployZipPackage', async (context: any) => {
+    });
+    initAsyncCommand(context,'appService.DeployZipPackage', async (context: any) => {
         if (context instanceof AppServiceNode) {
             const wizard = new WebAppZipPublisher(outputChannel, azureAccount, context.subscription, context.site);
             await wizard.run();
         } else if (context instanceof vscode.Uri) {
             const wizard = new WebAppZipPublisher(outputChannel, azureAccount, null, null, context.fsPath, null);
             await wizard.run();
         }
-    }));
-    context.subscriptions.push(vscode.commands.registerCommand('appService.ZipAndDeploy', async (context: any) => {
+    });
+    initAsyncCommand(context, 'appService.ZipAndDeploy', async (context: any) => {
         if (context instanceof vscode.Uri) {
             const folderPath = context.fsPath;
             const wizard = new WebAppZipPublisher(outputChannel, azureAccount, null, null, null, folderPath);
             await wizard.run();
         }
-    }));
+    });
 }
 
 export function deactivate() {
 }
+
+function initCommand(context: vscode.ExtensionContext, commandId: string, callback: (...args: any[]) => any) {
+    initAsyncCommand(context, commandId, (...args: any[]) => Promise.resolve(callback(...args)));
```

Comment:
I played with the parameters a bit and found out it was the calling of the 'callback' function - do 'callback(...args)' instead of 'callback(args)' fixes the problem and you can have more than one parameters passed to the callback.

Reply:
And I tested all the commands with this approach.

Reply:
Awesome, makes sense. I updated my PR to do the same